utils/post_utils.jsx

```diff
@@ -511,31 +512,50 @@ export function getValidUserPostFromId(state, id) {
     return getPost(state, id);
 }
 
+function isPostHeaderVisible(post, previousPost, getReplyCount) {
+    //If there is no previous post then current post must have visible header
+    if (!previousPost) {
+        return true;
+    }
+
+    //If current post is first message of a thread then it must have header visible
+    if (isFirstReply(post, previousPost)) {
+        return true;
+    }
+
+    //If current post is a comment and not the first comment of a thread then its header must be hidden
+    if (post.root_id) {
+        return false;
+    }
+
+    //If current post and previous posts both are not comments
+    if (!post.root_id && !previousPost.root_id) {
+        //If a non-comment post has some replies then its header must be visible
+        //If a non-comment post is a consecutive post then its header must be hidden
+        return getReplyCount(post) > 0 || !areConsecutivePostsBySameUser(post, previousPost);
+    }
+
+    return true;
+}
+
 export function getCurrentUserLastPostGroupFirstPostId(state, postListIds) {
```

Comment:
This name isn't correct anymore as replies break the post groups.

Rename to - `getLastCurrentUserPostWithHeaderId`